modules/widgets-panel/assets/js/editor/favorites-component.js

```diff
@@ -0,0 +1,140 @@
+import ComponentBase from 'elementor-api/modules/component-base';
+import * as commandsData from './commands-data/';
+import * as commands from './commands';
+import ignoreSectionWidgets from './helpers/ignoreSectionWidgets';
+
+export default class FavoritesComponent extends ComponentBase {
+	getNamespace() {
+		return 'panel-favorites';
+	}
+
+	defaultCommands() {
+		return this.importCommands( commands );
+	}
+
+	defaultData() {
+		return this.importCommands( commandsData );
+	}
+
+	registerAPI() {
+		super.registerAPI();
+		$e.routes.on( 'run:after', ( component, route ) => {
+			if ( 'panel/elements/categories' === route ) {
+				this.menuId = 'e-favorite-widget-context-menu';
+				this.favoriteWidgetMenu = null;
+				this.clickedWidget = null;
+
+				this.addFavoriteWidgetsEvents();
+				this.addToolTip();
+			}
+		} );
+	}
+
+	addToolTip() {
+		const favoriteCategory = jQuery( '#elementor-panel-category-favorites .elementor-panel-category-title > i' );
+		if ( favoriteCategory ) {
+			favoriteCategory.tipsy( {
+				title: function title() {
+					return this.getAttribute( 'data-tooltip' );
+				},
+				gravity: 's',
+			} );
+		}
+	}
+
+	addFavoriteWidgetsMenu( e ) {
+		// Remove the menu if exists
+		this.hideFavoriteWidgetsMenu();
+
+		const panel = jQuery( '#elementor-panel' );
+		// Find the clicked widget
+		this.clickedWidget = jQuery( e.target ).parents( '.elementor-element-wrapper' );
+		// Find the name (id) of clicked widget
+		this.clickedWidgetId = this.clickedWidget.find( '.elementor-element' ).data( 'id' );
+		// Find the category where user clicked
+		this.clickedWidgetCat = this.clickedWidget.parents( '.elementor-panel-category' ).attr( 'id' );
+		// Decided witch msg display. if clicked on widget under "Favorites" category display: Remove... otherwise Add...
+		const menuText = 'elementor-panel-category-favorites' === this.clickedWidgetCat ? 'Remove From Favorites' : 'Add To Favorites';
+		// Convert action text to class name
+		this.menuClass = menuText.toLowerCase().replaceAll( ' ', '-' );
+
+		// Create menu element with content
+		this.favoriteWidgetMenu = jQuery( '<div>', { id: this.menuId, class: this.menuClass } );
+		this.favoriteWidgetMenu.html( jQuery( '<span>' ).text( menuText ) );
```

Comment:
Create different elements for each menu, so that on click you don't need to guess what the user wants.

Reply:
Fixed